File: app/src/main/java/com/motorola/cdeives/motofretado/ViewMapPresenterLoader.java
package com.motorola.cdeives.motofretado;

import android.content.Context;
import android.support.annotation.MainThread;
import android.support.v4.content.Loader;
import android.util.Log;

public class ViewMapPresenterLoader extends Loader<ViewMapMvp.Presenter> {
    private static final String TAG = ViewMapPresenterLoader.class.getSimpleName();

    private ViewMapMvp.Presenter mPresenter;

    public ViewMapPresenterLoader(Context context) {
        super(context);
    }

    @Override
    @MainThread
    protected void onStartLoading() {
        Log.v(TAG, "> onStartLoading()");

        if (mPresenter == null) {
            forceLoad();
        } else {
            deliverResult(mPresenter);
        }

        Log.v(TAG, "< onStartLoading()");
    }

    @Override
    @MainThread
    protected void onForceLoad() {
        Log.v(TAG, "> onForceLoad()");

        mPresenter = new ViewMapPresenter(getContext());
        deliverResult(mPresenter);

        Log.v(TAG, "< onForceLoad()");
    }

    @Override
    @MainThread
    protected void onReset() {
        Log.v(TAG, "> onReset()");

        mPresenter = null;

        Log.v(TAG, "< onReset()");
    }
}

File: app/src/main/java/com/gmail/cristiandeives/motofretado/ViewMapPresenterLoader.kt
package com.gmail.cristiandeives.motofretado

import android.content.Context
import android.support.annotation.MainThread
import android.support.v4.content.Loader
import android.util.Log

@MainThread
internal class ViewMapPresenterLoader(context: Context) : Loader<ViewMapMvp.Presenter>(context) {
    companion object {
        private val TAG = ViewMapPresenterLoader::class.java.simpleName
    }

    private var mPresenter: ViewMapMvp.Presenter? = null

    override fun onStartLoading() {
        Log.v(TAG, "> onStartLoading()")

        if (mPresenter == null) {
            forceLoad()
        } else {
            deliverResult(mPresenter)
        }

        Log.v(TAG, "< onStartLoading()")
    }

    override fun onForceLoad() {
        Log.v(TAG, "> onForceLoad()")

        mPresenter = ViewMapPresenter(context)
        deliverResult(mPresenter)

        Log.v(TAG, "< onForceLoad()")
    }

    override fun onReset() {
        Log.v(TAG, "> onReset()")

        mPresenter = null

        Log.v(TAG, "< onReset()")
    }
}